### DISCUSSION TITLE
Montage with layout like Masonry JavaScript

### DISCUSSION THREAD

**Discussion by sbonaime** [ImageMagick/ImageMagick#2182] [Help]
Hi,
I would like to use montage or imagemagick (or montage) to combine 12 pictures with different sizes and orientation without taking account order but filling the gaps like the Masonry JavaScript like this example:

https://cloud.githubusercontent.com/assets/2784308/24494903/0c878852-1566-11e7-966e-054ffeb43377.png

How can I do that ? 
Thank you

**Comment by snibgo** [comment]
The problem is under-constrained. If the 12 inputs have different sizes, what decides how many rows the output should have? And the height of each row? Images per row? The order of images per row?

IM won't answer those questions.

**Reply by sbonaime** [reply]
Masonry JavaScript does this job. Each time this script is run, the layout could be different. In my question, I specifically wrote "without taking account order"

**Comment by urban-warrior** [comment]
We're working on a grid layout solution as you requested.  Its still about a month away before it will be released.

**Reply by musjj** [reply]
Did anything ever came out of this?

**Reply by urban-warrior** [reply]
Try the [Ashlar](https://imagemagick.org/script/formats.php#pseudo) pseudo-image format.

**Reply by musjj** [reply]
Thanks, it seems to be pretty close to what I want. Is it possible to explicitly specify the border size while keeping the resolution automatic?

**Reply by urban-warrior** [reply]
You can set the border by setting the image geometry as part of the output filename geometry.  For example:

```
magick -size 320x320 -label %f *.jpg  -resize 320x320 -define ashlar:tiles=30 'ashlar:ashlar.png[1920x1080+10+50]'
```

**Comment by fmw42** [comment]
You can append each row (using +smush X to add spacing), then crop to the width you want. The append (using -smush X) all the rows.
